kstreams/streams.py

```diff
@@ -132,20 +131,14 @@ async def stop(self) -> None:
                 if self.consumer is not None:
                     await self.consumer.stop()
 
-                if self._consumer_task is not None:
-                    self._consumer_task.cancel()
-
-            logger.info(
-                f"Stream consuming from topics {self.topics} has stopped!!! \n\n"
-            )
-
-    async def _subscribe(self) -> None:
-        # Always create a consumer on stream.start
-        self.consumer = self._create_consumer()
+                logger.info(
+                    f"Stream consuming from topics {self.topics} has stopped!!! \n\n"
+                )
 
-        # add the chech tp avoid `mypy` complains
-        if self.consumer is not None:
-            await self.consumer.start()
+    def subscribe(self) -> None:
+        if self.consumer is None:
```

Comment:
should this raise a warning if called again? can this create a problem if you stop the consumer, and then resume? is it still subscribed?

Reply:
It should not be a problem because we are not using `unsubscribe`. According to `aiokafka`, the `stop` method does not unsubscribe the consumer, it 

1. Commit last consumed message if autocommit enabled
2. Leave group if used Consumer Groups